### ISSUE TITLE
New CSV format of Tastyworks

### ISSUE THREAD

**Issue by andreasemer** [laroche/tastyworks-pnl#39]
The current version of the Pythonscript can no longer be used, the Tastyworks website only allows you to download the new CSV format. Are there plans to adapt the format?

**Comment by avion23** [comment]
Hello,
I have a similar project. I autogenerated a converter but haven't tried it out yet. I would appreciate your feedback.
https://github.com/avion23/tastyworksTaxes/blob/main/tastyworksTaxes/legacy.py

**Comment by quantx-heiko** [comment]
I have adjusted the code for supporting the new Tastytrade data format exported from the Tastytrade App CSV export.
@laroche are you interested in incorporating this into your code?

**Comment by laroche** [comment]
Yes, sure. I'd love to convert over to the new data format. Can you post your changes here on github or just
send them to my email address?

best regards,

Florian La Roche

**Comment by elsbrock** [comment]
@quantx-heiko is it available online?

**Comment by quantx-heiko** [comment]
@elsbrock  I just cloned the repo and did changes locally. So they're not online available yet.
@laroche  However I did a couple of more changes to your original code than required for just the CSV format change e.g. added multipliers for other Futures. I also adjusted code to handle dates from the CSV as date object during processing instead of pure stings.
If it is OK for you I'd create a fork and a pull request with the necessary changes for the new format.

**Comment by laroche** [comment]
Sounds like those additional changes should also be merged and extend the current version,
looking forward to see the changes.

**Comment by MiErnst** [comment]
As the format is the same as in #28, #28 can be closed as duplicate as it seems the work happens in this issue.

**Comment by quantx-heiko** [comment]
@laroche find my code changes in pull request https://github.com/laroche/tastyworks-pnl/pull/42

**Comment by quantx-heiko** [comment]
Issue #37 is also a duplicate of this issue and might be closed once the changes are incorporated.

**Comment by fdzaebel** [comment]
Hi all, does the new CSV format now work with this (great) tool? Thanks.

**Comment by alneven** [comment]
I have the same question. Previously, I just copied the new data to the top of last year’s data, ran the script, and got the numbers.
Now I'm wondering what the correct approach is to get continuous numbers.
Should I only download the data for 2024 and run the script with just that?
As far as I remember, I didn’t carry over any open positions from 2023 to 2024, so it should be fine.

**Comment by quantx-heiko** [comment]
I generated the tax report for 2024 with the current version based on the current Tastytrade export CSV file.
From my point of view this issue can be closed.
